packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/navigation_callback_delegate.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'webview_platform.dart';
+
+/// An interface defining navigation events that occur on the native platform.
+///
+/// The [WebViewControllerDelegate] is notifying this delegate on events that
+/// happened on the platform's webview. Platform implementations should
+/// implement this class and pass an instance to the [WebViewControllerDelegate].
+abstract class NavigationCallbackDelegate extends PlatformInterface {
```

Comment:
I missed this before, but this should be called `NavigationCallbackHandlerDelegate`. Mainly because `NavigationCallback` wouldn't work well for as app-facing class.

Reply:
Renamed this to `PlatformNavigationCallbackDelegate` as per our off-line discussion on 05/04/2022.